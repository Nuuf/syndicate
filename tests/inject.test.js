/* eslint-disable @typescript-eslint/no-var-requires */
const syndicate = require('../dist');

test('changes the parent of an entity using a sibling', () => {
  const root = syndicate.createRoot();
  syndicate.add(root, syndicate.create({}, 'parent'));
  const a = syndicate.add(root, syndicate.create({}, 'siblingA'), syndicate.find(root, 'parent'));
  const b = syndicate.add(root, syndicate.create({}, 'siblingB'));
  syndicate.inject(root, a, b, syndicate.AFTER);
  syndicate.inject(root, a, b, syndicate.BEFORE);
  syndicate.inject(root, a, b, syndicate.START);
  syndicate.inject(root, a, b, syndicate.END);
  expect(JSON.stringify(root, null, 2)).toMatchSnapshot();
});
