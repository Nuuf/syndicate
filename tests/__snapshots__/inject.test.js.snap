// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the parent of an entity using a sibling 1`] = `
"[
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":-1,\\\\\\"parentId\\\\\\":null,\\\\\\"index\\\\\\":0,\\\\\\"id\\\\\\":\\\\\\"root\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"name\\\\\\":\\\\\\"root\\\\\\"},\\\\\\"childrenIndices\\\\\\":[1,3,2],\\\\\\"childrenIds\\\\\\":[\\\\\\"parent\\\\\\",\\\\\\"siblingB\\\\\\",\\\\\\"siblingA\\\\\\"]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":0,\\\\\\"parentId\\\\\\":\\\\\\"root\\\\\\",\\\\\\"index\\\\\\":1,\\\\\\"id\\\\\\":\\\\\\"parent\\\\\\",\\\\\\"data\\\\\\":{},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":0,\\\\\\"parentId\\\\\\":\\\\\\"root\\\\\\",\\\\\\"index\\\\\\":2,\\\\\\"id\\\\\\":\\\\\\"siblingA\\\\\\",\\\\\\"data\\\\\\":{},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":0,\\\\\\"parentId\\\\\\":\\\\\\"root\\\\\\",\\\\\\"index\\\\\\":3,\\\\\\"id\\\\\\":\\\\\\"siblingB\\\\\\",\\\\\\"data\\\\\\":{},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\"
]"
`;
